include/nil/crypto3/zk/commitments/polynomial/lpc.hpp

```diff
@@ -188,28 +204,23 @@ namespace nil {
                         const std::map<std::size_t, commitment_type> &commitments,
                         transcript_type &transcript
                     ) {
-                        for (auto const&[b_ind, fixed]: _batch_fixed) {
-                            if(!fixed) continue;
-                            this->append_eval_point(b_ind, _etha);
-                            for( std::size_t i = 0; i < proof.z.get_batch_size(b_ind); i++) {
-                                if(this->_fixed_polys_values[b_ind][i] != proof.z.get(b_ind, i, proof.z.get_poly_points_number(b_ind, i) - 1)) {
-                                    return false;
-                                }
-                            }
-                        }
-
                         this->_z = proof.z;
                         for (auto const &it: commitments) {
                             transcript(commitments.at(it.first));
                         }
 
                         auto points = this->get_unique_points();
+
                         // List of unique eval points set. [id=>points]
-                        typename std::vector<typename field_type::value_type> U(points.size());
+                        std::size_t total_points = points.size();
+                        for( std::size_t i = 0; i < _batch_fixed.size(); i++){
```

Comment:
You can re-write this as this:

if (std::any_of(_batch_fixed.begin(), _batch_fixed.end(), [](auto i){return i != 0;}))
   total_points++;